File: demo/src/main/java/com/maple/demo/MainActivity.java
package com.maple.demo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.maple.demo.databinding.ActivityMainBinding;
import com.maple.tools.ToolsMainActivity;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, ToolsMainActivity.class);
                startActivity(intent);

//                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
//                        .setAction("Action", null).show();
            }
        });
    }

}

File: demo/src/main/java/com/maple/demo/MainActivity.kt
package com.maple.demo

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.maple.demo.databinding.ActivityMainBinding
import com.maple.tools.base.BaseActivity

/**
 * 工具集主页面
 */
class MainActivity : BaseActivity() {
    private val adapter by lazy {
        ToolsAdapter(mContext).apply {
            setOnItemClickListener { item, _ ->
                startActivity(Intent(this@MainActivity, item.clsName))
            }
            setOnItemLongClickListener { _, position ->
                val record = getItem(position)
                Toast.makeText(mContext, "long click ${record.title}", Toast.LENGTH_SHORT).show()
                true
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        initUI(binding)
    }

    private fun initUI(binding: ActivityMainBinding) {
        with(binding) {
            with(topBar) {
                ivBack.visibility = View.GONE
                tvTitle.text = "开发工具集"
            }
            rvData.adapter = adapter
        }
        adapter.refreshData(Constant.toolData)
    }

}